src/component/DefaultTooltipContent.tsx

```diff
@@ -81,21 +84,24 @@ export class DefaultTooltipContent<TValue extends ValueType, TName extends NameT
           ...itemStyle,
         };
         const finalFormatter = entry.formatter || formatter || defaultFormatter;
-        let { value, name } = entry;
-        if (finalFormatter && value != null && name != null) {
+        const { value, name } = entry;
+        let finalValue: React.ReactNode = value;
+        let finalName: React.ReactNode = name;
+        if (finalFormatter && finalValue != null && finalName != null) {
           const formatted = finalFormatter(value, name, entry, i, payload);
           if (Array.isArray(formatted)) {
-            [value, name] = formatted as [TValue, TName];
+            [finalValue, finalName] = formatted;
           } else {
-            value = formatted;
+            finalValue = formatted;
           }
         }
+
         return (
           // eslint-disable-next-line react/no-array-index-key
           <li className="recharts-tooltip-item" key={`tooltip-item-${i}`} style={finalItemStyle}>
-            {isNumOrStr(name) ? <span className="recharts-tooltip-item-name">{name}</span> : null}
-            {isNumOrStr(name) ? <span className="recharts-tooltip-item-separator">{separator}</span> : null}
-            <span className="recharts-tooltip-item-value">{value}</span>
+            {isNumOrStr(finalName) ? <span className="recharts-tooltip-item-name">{finalName}</span> : null}
+            {isNumOrStr(finalName) ? <span className="recharts-tooltip-item-separator">{separator}</span> : null}
```

Comment:
these checks prevent a react element from being rendered here

Reply:
You're right, I think I should revert the name type back to `TName` type, I'd love your feedback here too
https://github.com/recharts/recharts/pull/3618#discussion_r1224190579